docs/cars.py

```diff
@@ -0,0 +1,114 @@
+#!/usr/bin/env python3
+import jinja2
+import os
+from collections import defaultdict
+from sortedcontainers import SortedList
+
+from common.basedir import BASEDIR
+from common.params import Params
+from selfdrive.car.car_helpers import interfaces, get_interface_attr
+from selfdrive.car.docs import Tier, Column, StarColumns, MAKES_GOOD_STEERING_TORQUE, get_footnote, get_star_icon
+from selfdrive.car.hyundai.radar_interface import RADAR_START_ADDR as HKG_RADAR_START_ADDR
+from selfdrive.test.test_routes import non_tested_cars
+
+
+def get_all_footnotes():
+  all_footnotes = {}
+  i = 1
+  for _, footnotes in get_interface_attr("Footnote").items():
+    if footnotes is not None:
+      for footnote in footnotes:
+        all_footnotes[footnote] = i
+        i += 1
+  return all_footnotes
+
+
+ALL_FOOTNOTES = get_all_footnotes()
+
+
+class Car:
+  def __init__(self, car_info, CP):
+    self.make, self.model = car_info.name.split(' ', 1)
+    self.row, star_count = self.get_row(car_info, CP)
+    self.tier = {5: Tier.GOLD, 4: Tier.SILVER}.get(star_count, Tier.BRONZE)
+
+  def get_row(self, car_info, CP):
+    # TODO: add YouTube videos
+    # Some minimum steering speeds are not yet in CarParams
+    min_steer_speed = CP.minSteerSpeed
+    if car_info.min_steer_speed is not None:
+      min_steer_speed = car_info.min_steer_speed
+      assert CP.minSteerSpeed == 0, "Minimum steer speed set in both CarInfo and CarParams for {}".format(CP.carFingerprint)
+
+    min_enable_speed = CP.minEnableSpeed
+    if car_info.min_enable_speed is not None:
+      min_enable_speed = car_info.min_enable_speed
+
+    stars = [CP.openpilotLongitudinalControl and not CP.radarOffCan, min_enable_speed <= 0., min_steer_speed <= 0.,
+             CP.carName in MAKES_GOOD_STEERING_TORQUE, CP.carFingerprint not in non_tested_cars]
+    row = [self.make, self.model, car_info.package, *map(lambda star: "full" if star else "empty", stars)]
+
+    # Check for car footnotes and star demotions
+    star_count = 0
+    for row_idx, column in enumerate(Column):
+      footnote = get_footnote(car_info, column)
+      if column in StarColumns:
+        # Demote if footnote specifies a star
+        if footnote is not None and footnote.value.star is not None:
+          row[row_idx] = footnote.value.star
+        star_count += row[row_idx] == "full"
+        row[row_idx] = get_star_icon(row[row_idx])
+
+      if footnote is not None:
+        row[row_idx] += "<sup>{}</sup>".format(ALL_FOOTNOTES[footnote])
```

Comment:
can we make the footnote superscript link to the footnote?

Reply:
Done, but the star images are clickable and land you to the top of the page, so this might be frustrating for some users on small screens/mobile